### PR TITLE
Add System.Diagnostics.Debug API docs

### PR DESCRIPTION
**Pull request by tommcdon**
## Summary

Add System.Diagnostics.Debug API documentation for missing triple slash docs

Partially addresses #dotnet/runtime/issues/58654

### CODE REVIEW

File: xml/System.Diagnostics/Debug.xml [hunk 1]
```diff
@@ -263,10 +263,51 @@
...
+```  
+  
+   
+  
+## Examples  
+ The following example checks whether the `type` parameter is valid. If `type` is `null`, <xref:System.Diagnostics.Trace.Assert%2A> outputs a message.  
+  
+ :::code language="cpp" source="~/samples/snippets/cpp/VS_Snippets_CLR_Classic/classic Debug.Assert1 Example/CPP/source.cpp" id="Snippet1":::
+ :::code language="csharp" source="~/samples/snippets/csharp/VS_Snippets_CLR_Classic/classic Debug.Assert1 Example/CS/source.cs" id="Snippet1":::
+ :::code language="vb" source="~/samples/snippets/visualbasic/VS_Snippets_CLR_Classic/classic Debug.Assert1 Example/VB/source.vb" id="Snippet1":::
```
Comment (carlossanlop) [inline xml/System.Diagnostics/Debug.xml:299]: I see the examples existed in older `Assert` overloads. But I think they are relevant to those overloads, not to this one: they don't show how to use the new `ref System.Diagnostics.Debug.AssertInterpolatedStringHandler message` parameter.

File: xml/System.Diagnostics/Debug.xml [hunk 2]
```diff
@@ -2826,10 +2972,56 @@ Debug.WriteIf(mySwitch.TraceError, "aNumber = " + aNumber + " out of range");
...
+if(mySwitch.TraceError)   
+    Debug.WriteLine("aNumber = " + aNumber + " out of range");  
+```  
+  
+ Second example:  
+  
+```csharp  
+Debug.WriteLineIf(mySwitch.TraceError, "aNumber = " + aNumber + " out of range");  
+```</para>
+        </block>
```
Comment (carlossanlop) [inline xml/System.Diagnostics/Debug.xml:3016]: This block also seems only relevant to the older `WriteLineIf` overload, but not to this new overload that adds a `ref System.Diagnostics.Debug.WriteIfInterpolatedStringHandler message` parameter.

**Review by carlossanlop — APPROVED** [review]
Looks good. The only thing is my concern about adding examples that were relevant for the older overloads, but not for the new ones. Maybe those sections should be removed.

The remark warnings are ok, those apply to all overloads.

File: xml/System.Diagnostics/Debug.xml [hunk 3]
```diff
@@ -263,10 +263,41 @@
...
+  
+> [!NOTE]
+>  The display of the message box depends on the presence of the <xref:System.Diagnostics.DefaultTraceListener>. If the <xref:System.Diagnostics.DefaultTraceListener> is not in the <xref:System.Diagnostics.Trace.Listeners%2A> collection, the message box is not displayed. The <xref:System.Diagnostics.DefaultTraceListener> can be removed by the [&lt;clear&gt;](/dotnet/framework/configure-apps/file-schema/trace-debug/clear-element-for-listeners-for-trace), the [&lt;remove&gt;](/dotnet/framework/configure-apps/file-schema/trace-debug/remove-element-for-listeners-for-trace), or by calling the <xref:System.Diagnostics.TraceListenerCollection.Clear%2A> method on the <xref:System.Diagnostics.Trace.Listeners%2A> property (`System.Diagnostics.Trace.Listeners.Clear()`).
```
Comment (gewarren) [inline xml/System.Diagnostics/Debug.xml:280]:
```suggestion
> The display of the message box depends on the presence of the <xref:System.Diagnostics.DefaultTraceListener>. If the <xref:System.Diagnostics.DefaultTraceListener> is not in the <xref:System.Diagnostics.Trace.Listeners%2A> collection, the message box is not displayed. The <xref:System.Diagnostics.DefaultTraceListener> can be removed by the [&lt;clear&gt; element](/dotnet/framework/configure-apps/file-schema/trace-debug/clear-element-for-listeners-for-trace) or the [&lt;remove&gt; element](/dotnet/framework/configure-apps/file-schema/trace-debug/remove-element-for-listeners-for-trace), or by calling the <xref:System.Diagnostics.TraceListenerCollection.Clear%2A> method on the <xref:System.Diagnostics.Trace.Listeners%2A> property (`System.Diagnostics.Trace.Listeners.Clear()`).  
```

File: xml/System.Diagnostics/Debug.xml [hunk 4]
```diff
@@ -421,11 +452,42 @@
...
```
Comment (gewarren) [inline xml/System.Diagnostics/Debug.xml:470]:
```suggestion
> The display of the message box depends on the presence of the <xref:System.Diagnostics.DefaultTraceListener>. If the <xref:System.Diagnostics.DefaultTraceListener> is not in the <xref:System.Diagnostics.Trace.Listeners%2A> collection, the message box is not displayed. The <xref:System.Diagnostics.DefaultTraceListener> can be removed by the [&lt;clear&gt; element](/dotnet/framework/configure-apps/file-schema/trace-debug/clear-element-for-listeners-for-trace) or the [&lt;remove&gt; element](/dotnet/framework/configure-apps/file-schema/trace-debug/remove-element-for-listeners-for-trace), or by calling the <xref:System.Diagnostics.TraceListenerCollection.Clear%2A> method on the <xref:System.Diagnostics.Trace.Listeners%2A> property (`System.Diagnostics.Trace.Listeners.Clear()`).  
```

**Review by gewarren — APPROVED** [review]

**Pull request merged**